tests/osquery/test_api_views.py

```diff
@@ -2401,3 +2764,283 @@ def test_delete_query_permission_denied(self):
+        self.assertEqual(configuration_pack.tags.count(), 1)
+        self.assertEqual(configuration_pack.pack, new_pack)
+
+    # create configuration pack
+
+    def test_create_configurationpack_unauthorized(self):
+        response = self.post_json_data(reverse("osquery_api:configuration_packs"), {}, include_token=False)
+        self.assertEqual(response.status_code, 401)
+
+    def test_create_configurationpack_permission_denied(self):
+        response = self.post_json_data(reverse("osquery_api:configuration_packs"), {})
+        self.assertEqual(response.status_code, 403)
+
+    def test_create_configurationpack_fields_empty(self):
+        self.set_permissions("osquery.add_configurationpack")
+        response = self.post_json_data(reverse("osquery_api:configuration_packs"), {})
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json(), {
+            "configuration": ["This field is required."],
+            "pack": ["This field is required."]
+        })
+
+    def test_create_configurationpack_conflict(self):
+        self.set_permissions("osquery.add_configurationpack")
+        configuration_pack = self.force_configuration_pack(force_tags=True)
+        data = {
+            "configuration": configuration_pack.configuration.pk,
+            "pack": configuration_pack.pack.pk,
+        }
+        response = self.post_json_data(reverse("osquery_api:configuration_packs"), data)
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json(), {
+            "non_field_errors": ["The fields configuration, pack must make a unique set."]
+        })
+
+    def test_create_configurationpack_configuration_with_multiple_packs(self):
+        self.set_permissions("osquery.add_configurationpack")
+        configuration = self.force_configuration()
+        packs = [self.force_pack().pk for i in range(0, 3)]
+        for pack in packs:
+            data = {
+                "configuration": configuration.pk,
+                "pack": pack
+            }
+            response = self.post_json_data(reverse("osquery_api:configuration_packs"), data)
+            self.assertEqual(response.status_code, 201)
+            configuration_pack = ConfigurationPack.objects.filter(
+                configuration_id=configuration.pk,
+                pack_id=pack
+            ).first()
+            self.assertEqual(response.json(), {
+                "id": configuration_pack.pk,
+                "configuration": configuration.pk,
+                "tags": [],
+                "pack": pack
+            })
+            self.assertEqual(configuration_pack.pack.pk, pack)
+            self.assertEqual(configuration_pack.configuration.pk, configuration.pk)
+        result = ConfigurationPack.objects.filter(configuration_id=configuration.pk).count()
+        self.assertEqual(result, 3)
+
+    def test_create_configurationpack_with_pack_id_used_in_another_configuration_pack(self):
```

Comment:
Maybe find a better name for the test, to describe that you want to make sure that adding the same pack to different configurations is possible.